loader/normalize.go

```diff
@@ -0,0 +1,43 @@
+package loader
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/score-spec/score-go/types"
+)
+
+// Normalize normalizes the target Workload by:
+// * embedding container file sources as content
+func Normalize(w *types.Workload, baseDir string) error {
+	for name, c := range w.Containers {
+		for i, f := range c.Files {
+			if f.Source != nil {
+				raw, err := readFile(baseDir, *f.Source)
+				if err != nil {
+					return fmt.Errorf("embedding file '%s' for container '%s': %w", *f.Source, name, err)
+				}
+
+				c.Files[i].Source = nil
+				c.Files[i].Content = &raw
+			}
+		}
+	}
+
+	return nil
+}
+
+// readFile reads a text file into memory
+func readFile(baseDir, path string) (string, error) {
+	if !filepath.IsAbs(path) {
+		path = filepath.Join(baseDir, path)
+	}
+
+	raw, err := os.ReadFile(path)
+	if err != nil {
+		return "", fmt.Errorf("reading '%s': %w", path, err)
```

Comment:
nit pick: the error here usually contains the file path and operation already, as seen in the unit test, it may be cleaner just to return the inner error here.